globalconf_test.go

```diff
@@ -96,6 +96,27 @@ func TestParse_GlobalWithDottedFlagname(t *testing.T) {
 	}
 }
 
+func TestParse_GlobalWithEmptyValue(t *testing.T) {
+	t.Log("Given the need to test overwriting a flag with an environmental variable set to an empty string")
```

Comment:
I picked up this test explaining style from Go In Action. I saw  a number of benefits to it such as when you `go test -v` the output looks like:
```
=== RUN   TestParse_GlobalWithEmptyValue
--- PASS: TestParse_GlobalWithEmptyValue (0.00s)
        globalconf_test.go:100: Given the need to test overwriting a flag with an environmental variable set to an empty string
        globalconf_test.go:101:         Reset the flags and clear the env
        globalconf_test.go:106:         Set the env var for the c flag to an empty string
        globalconf_test.go:108:         Define the c flag and parse the INI file
        globalconf_test.go:113:                 Expect flagC to an empty string and not be the value Hello World from the INI file
```

This helped me see where failures were happening more quickly along with educating me on what the test was doing.